test/e2e/flowaggregator_test.go

```diff
@@ -238,59 +262,76 @@ func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP stri
+					}
 				} else {
-					checkPodAndNodeData(t, record, "perftest-a", controlPlaneNodeName(), "perftest-c", workerNodeName(1))
-					checkFlowType(t, record, ipfixregistry.FlowTypeInterNode)
-				}
-
-				if checkService {
-					if isIntraNode {
-						if !strings.Contains(record, "antrea-test/perftest-b") {
-							t.Errorf("Record with ServiceIP does not have Service name")
-						}
-					} else {
-						if !strings.Contains(record, "antrea-test/perftest-c") {
-							t.Errorf("Record with ServiceIP does not have Service name")
-						}
+					if !strings.Contains(record, "antrea-test/perftest-c") {
+						t.Errorf("Record with ServiceIP does not have Service name")
 					}
 				}
-				if checkNetworkPolicy {
-					// Check if records have both ingress and egress network policies.
-					if !strings.Contains(record, ingressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with ingress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
-					}
-					if !strings.Contains(record, egressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with egress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
-					}
+			}
+			if checkNetworkPolicy {
+				// Check if records have both ingress and egress network policies.
+				if !strings.Contains(record, ingressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with ingress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
+				}
+				if !strings.Contains(record, egressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with egress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
+				}
+			}
+			if checkBandwidth && !strings.Contains(record, "octetDeltaCount: 0") {
+				exportTime := int64(getUnit64FieldFromRecord(t, record, "flowEndSeconds"))
+				curOctetTotalCount := getUnit64FieldFromRecord(t, record, "octetTotalCountFromSourceNode")
+				if curOctetTotalCount > octetTotalCount {
+					octetTotalCount = curOctetTotalCount
 				}
-				// Check the bandwidth using octetDeltaCount in data record.
-				if checkBandwidth {
-					checkBandwidthFromRecord(t, record, bandwidth)
+				curOctetDeltaCount := getUnit64FieldFromRecord(t, record, "octetDeltaCountFromSourceNode")
+				// Check the bandwidth using octetDeltaCountFromSourceNode if this record is not the first and not the last one.
```

Comment:
".., if this record is not either the first record or the last in the stream of records."

Reply:
Thanks, addressed.